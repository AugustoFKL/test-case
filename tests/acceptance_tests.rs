#![cfg(test)]

#[macro_use]
extern crate lazy_static;

mod acceptance {
    use std::process::{Command, Output};

    fn run_tests() -> Output {
        Command::new("cargo")
            .args(&["test", "test_cases"])
            .output()
            .expect("cargo command failed to start")
    }

    lazy_static! {
        static ref ACTUAL: String = {
            let output = run_tests().stdout;

            String::from_utf8_lossy(&output).to_string()
        };
    }

    fn actual<'a>() -> &'a str {
        ACTUAL.as_ref()
    }

    #[test]
    fn runs_all_test_cases() {
        assert!(actual().contains("running 29 tests"));
    }

    #[test]
    fn escapes_unnecessary_leading_underscore() {
        assert!(actual().contains("test test_cases::leading_underscore_in_test_name::dummy ... ok"));
    }

    #[test]
    fn escapes_names_starting_with_digit() {
        assert!(actual().contains("test test_cases::basic_test::_1 ... ok"));
    }

    #[test]
    fn removes_repeated_underscores() {
        assert!(actual().contains("test test_cases::arg_expressions::_2_4_6_to_string ... ok"));
    }

    #[test]
    fn escapes_rust_keywords() {
        assert!(actual().contains("test test_cases::keyword_test::_true ... ok"));
    }
}