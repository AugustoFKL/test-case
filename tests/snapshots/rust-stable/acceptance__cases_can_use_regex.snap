---
source: tests/acceptance_tests.rs
expression: output
---
error: incomplete escape sequence, reached end of pattern prematurely
error: test failed, to rerun pass `--lib`
test regex_test::_abcabc201_expects_complex_regex_r_d_ ... ok
test regex_test::_abcabc201_expects_complex_regex_r_d_4_ ... FAILED
test regex_test::_abcabc_expects_complex_regex_r_abc_ ... ok
test regex_test::_kumkwat_expects_complex_regex_r_ ... FAILED
test regex_test::_kumkwat_expects_complex_regex_r_abc_ ... FAILED
test regex_test::_kumkwat_expects_complex_regex_r_kumkwat_ ... ok
test result: FAILED. 3 passed; 3 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
thread 'regex_test::_abcabc201_expects_complex_regex_r_d_4_' panicked at src/lib.rs:5:1:
thread 'regex_test::_kumkwat_expects_complex_regex_r_' panicked at src/lib.rs:5:1:
thread 'regex_test::_kumkwat_expects_complex_regex_r_abc_' panicked at src/lib.rs:5:1:
